### PR TITLE
Close exercise popup on ESC click

### PR DESCRIPTION
**Pull request by olexandrd**

### CODE REVIEW

File: src/js/exercisePopUp.js [hunk 1]
```diff
@@ -5,15 +5,27 @@ import iziToast from 'izitoast';
...
+        'click',
+        handleAddToFavorites,
+        { passive: true }
+      );
+      refs.exercisePopUpBackdrop.addEventListener(
+        'click',
+        handleBackdropClick,
+        { passive: true }
+      );
+      document.addEventListener('keydown', handleESCClick);
```
Comment (olexandrd) [inline src/js/exercisePopUp.js:28]: this was formated by pretifier

Comment (olexandrd) [reply src/js/exercisePopUp.js:28]: actual new line `document.addEventListener('keydown', handleESCClick);`

File: src/js/exercisePopUp.js [hunk 2]
```diff
@@ -26,9 +38,13 @@ function closeExercisePopUp() {
   refs.exercisePopUpBackdrop.classList.remove('is-open');
   refs.body.classList.remove('overflow-hidden');
   // Remove all event listeners
-  refs.exercisePopupAddFavoritesBtn.removeEventListener('click', handleAddToFavorites);
+  refs.exercisePopupAddFavoritesBtn.removeEventListener(
+    'click',
+    handleAddToFavorites
+  );
```
Comment (olexandrd) [inline src/js/exercisePopUp.js:44]: was formatting by pretifier

File: src/js/exercisePopUp.js [hunk 3]
```diff
@@ -68,7 +84,15 @@ function addExerciseToFavorites(exercise) {
 }
 
 function handleAddToFavorites(event) {
-  const exercise = JSON.parse(refs.exercisePopupAddFavoritesBtn.dataset.exercise);
+  const exercise = JSON.parse(
+    refs.exercisePopupAddFavoritesBtn.dataset.exercise
+  );
```
Comment (olexandrd) [inline src/js/exercisePopUp.js:89]: this was formated by pretifier

File: src/js/exercisePopUp.js [hunk 4]
```diff
@@ -68,7 +84,15 @@ function addExerciseToFavorites(exercise) {
...
   addExerciseToFavorites(exercise);
   event.stopPropagation();
 }
+
+function handleESCClick(e) {
+  if (e.code === 'Escape') {
+    closeExercisePopUp();
+  }
+}
```
Comment (olexandrd) [inline src/js/exercisePopUp.js:98]: actual new code

File: src/js/exercisePopUp.js [hunk 5]
```diff
@@ -26,9 +38,13 @@ function closeExercisePopUp() {
...
   refs.body.classList.remove('overflow-hidden');
   // Remove all event listeners
-  refs.exercisePopupAddFavoritesBtn.removeEventListener('click', handleAddToFavorites);
+  refs.exercisePopupAddFavoritesBtn.removeEventListener(
+    'click',
+    handleAddToFavorites
+  );
   refs.exercisePopUpBackdrop.removeEventListener('click', handleBackdropClick);
   refs.exercisePopUpCloseBtn.removeEventListener('click', closeExercisePopUp);
+  document.removeEventListener('keydown', handleESCClick);
```
Comment (olexandrd) [inline src/js/exercisePopUp.js:47]: actual new code

**Review by kazamov — APPROVED** [review]

**Pull request merged**